packages/containers/src/ConfirmDialog/ConfirmDialog.container.js

```diff
@@ -0,0 +1,51 @@
+import React, { PropTypes } from 'react';
+import { Map } from 'immutable';
+import { ConfirmDialog as Component } from 'react-talend-components';
+
+import { getActionsProps } from '../actionAPI';
+import { statePropTypes, initState } from '../state';
+
+export const DEFAULT_STATE = new Map({
+	show: false,
+	validateAction: {
+		label: 'Ok',
+		bsStyle: 'primary',
+		onClick: () => {},
+	},
+	cancelAction: {
+		label: 'No',
+		onClick: () => {},
+	},
+});
+
+class ConfirmDialog extends React.Component {
+	static displayName = 'CMFContainer(ConfirmDialog)';
+	static propTypes = {
+		...statePropTypes,
+	};
+
+	static contextTypes = {
+		store: PropTypes.object,
+		registry: PropTypes.object,
+	};
+
+	componentDidMount() {
+		initState(this.props);
+	}
+
+	render() {
+		const state = (this.props.state || DEFAULT_STATE).toJS();
+		// looking for the given action setting
+		// and adding onClick event using the retrieved  action creator
+		state.validateAction = getActionsProps(this.context, state.validateAction, state.model);
+		state.cancelAction = getActionsProps(this.context, state.cancelAction, state.model);
```

Comment:
yeah in fact they are string props in the end when the host app change the state to update validateAction and cancelAction. The issue now if we remove the added code in ActionAPI and make the props as string is that we'll need to put existing action id in these props, else it will end in error when calling getActionInfo. 
The benefit of having validateAction as object in DEFAULT_STATE is that it won't return propTypes warning and we see clearly what the component needs as an input.